### PR TITLE
[ENH] FreshPRINCERegressor, RotationForestRegressor and minor changes to FreshPRINCEClassifier

### PR DESCRIPTION
**Pull request by dguijo**
#### Reference Issues/PRs

None

#### What does this implement/fix? Explain your changes.

- Modifies the names of FreshPRINCE and RotationForest classifiers to FreshPRINCEClassifier and RotationForestClassifier. 
- Adds the RotationForestRegressor.
- Adds the FreshPRINCERegressor.
- Modifies some of the previous tests for time series regression.
- Adds the smallest multivariate dataset (CardanoSentiment).
- Modifies the load of the Covid3Month dataset.

#### Does your contribution introduce a new dependency? If yes, which one?

It doesn't. 

#### What should a reviewer concentrate their feedback on?

- [] Check that I've modified all old references to FreshPRINCE and RotationForest. 
- [] Check that the tests added for the FreshPRINCEClassifier and FreshPRINCERegressor are correct.
- [] Check that functions written for loading new TSER datasets are correct. 

#### Did you add any tests for the change?

Yes, for FreshPRINCEClassifier and FreshPRINCERegressor.

##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/aeon-toolkit/aeon/blob/main/.all-contributorsrc).
- [x] Optionally, I've updated aeon's [CODEOWNERS](https://github.com/aeon-toolkit/aeon/blob/main/CODEOWNERS) to receive notifications about future changes to these files.
- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG] indicating whether the PR topic is related to enhancement, maintenance, documentation, or bug.
- [] For new estimators, I've added the estimator to the online documentation.

### CODE REVIEW

**Review by MatthewMiddlehurst — COMMENTED** [review]
Nice contributions David, thanks! 

Only a couple of things I think should be changed.

File: aeon/classification/feature_based/__init__.py [hunk 1]
```diff
@@ -11,11 +11,13 @@
...
     "TSFreshClassifier",
-    "FreshPRINCE",
+    "FreshPRINCEClassifier",
 ]
 
 from aeon.classification.feature_based._catch22_classifier import Catch22Classifier
-from aeon.classification.feature_based._fresh_prince import FreshPRINCE
+from aeon.classification.feature_based._fresh_prince_classifier import (
+    FreshPRINCEClassifier,
+)
```
[the remark this answers is not in the document]
Comment (dguijo) [reply aeon/classification/feature_based/__init__.py:20]: Ok, you're right Matt. I'll do asap. Thanks for reviewing it. :smile:

File: aeon/datasets/_single_problem_loaders.py [hunk 2]
```diff
@@ -1052,6 +1052,17 @@ def load_covid_3month(split=None, return_X_y=True):
...
+        Memory data format specification to return X in, None = "nested_univ" type.
+        str can be any supported aeon Panel mtype,
+            for list of mtypes, see datatypes.MTYPE_REGISTER
+            for specifications, see examples/AA_datatypes_and_datasets.ipynb
+        commonly used specifications:
+            "nested_univ: nested pd.DataFrame, pd.Series in cells
+            "numpy3D"/"numpy3d"/"np3D": 3D np.ndarray (instance, variable, time index)
+            "numpy2d"/"np2d"/"numpyflat": 2D np.ndarray (instance, time index)
+            "pd-multiindex": pd.DataFrame with 2-level (instance, time) MultiIndex
+        Exception is raised if the data cannot be stored in the requested type.
```
[the remark this answers is not in the document]
Comment (dguijo) [reply aeon/datasets/_single_problem_loaders.py:1065]: In this case, I added the return_type in the load_covid_3month [for the rotation_forest_regressor tests](https://github.com/aeon-toolkit/aeon/pull/384/commits/792a2248d3d8cc0e671554e982614c53ace5e83d#diff-3139127c27803331bffa934bbbe31f524f0f803932883a8619adadd1aa0a14fa).

I think this is done in the same way for the [tests of the rotation_forest classifier](https://github.com/aeon-toolkit/aeon/blob/main/aeon/classification/sklearn/tests/test_rotation_forest.py#L12). 

I don't mind doing it in a different way if you prefer so, Matt.

[the remark this answers is not in the document]
Comment (dguijo) [reply aeon/datasets/_single_problem_loaders.py:1065]: ah, yeah, I see. I'll do too, completely agree. Thank you, Matt.

**Pull request merged**